android/src/com/mapswithme/maps/traffic/TrafficManager.java

```diff
@@ -0,0 +1,181 @@
+  }
+
+  public void disable()
+  {
+    mLogger.d("Disable traffic");
+    TrafficState.nativeDisable();
+  }
+
+  public boolean isEnabled()
+  {
+    return mState != TrafficState.DISABLED;
+  }
+
+  public void attach(@NonNull TrafficCallback callback)
+  {
+    if (mCallbacks.contains(callback))
+    {
+      throw new IllegalStateException("A callback '" + callback
+                                      + "' is already attached. Check that the 'detach' method was called.");
+    }
+
+    mLogger.d("Attach callback '" + callback + "'");
+    mCallbacks.add(callback);
+    postPendingState();
+  }
+
+  private void postPendingState()
+  {
+    mStateChangeListener.onTrafficStateChanged(mState);
+  }
+
+  public void detach()
+  {
+    if (mCallbacks.isEmpty())
+    {
+      throw new IllegalStateException("There are no attached callbacks. Invoke the 'detach' method " +
+                                      "only when it's really needed!");
+    }
+
+    for (TrafficCallback callback : mCallbacks)
+      mLogger.d("Detach callback '" + callback + "'");
+    mCallbacks.clear();
+  }
+
+  private class TrafficStateListener implements TrafficState.StateChangeListener
+  {
+    @Override
+    @MainThread
+    public void onTrafficStateChanged(@TrafficState.Value int state)
+    {
+      mLogger.d("onTrafficStateChanged current state = " + TrafficState.nameOf(mState)
+                + " new value = " + TrafficState.nameOf(state));
+      mState = state;
+
+      if (mCallbacks.isEmpty())
+        return;
+
+      iterateOverCallbacks(new Utils.Proc<TrafficCallback>()
+      {
+        @Override
+        public void invoke(@NonNull TrafficCallback callback)
+        {
+          switch (mState)
+          {
+            case TrafficState.DISABLED:
+              callback.onDisabled();
+              break;
+
+            case TrafficState.ENABLED:
+              callback.onEnabled();
+              break;
+
+            case TrafficState.WAITING_DATA:
+              callback.onWaitingData();
+              break;
+
+            case TrafficState.NO_DATA:
+              callback.onNoData();
+              break;
+
+            case TrafficState.OUTDATED:
+              callback.onOutdated();
+              break;
+
+            case TrafficState.NETWORK_ERROR:
+              callback.onNetworkError();
+              break;
+
+            case TrafficState.EXPIRED_DATA:
+              callback.onExpiredData();
+              break;
+
+            case TrafficState.EXPIRED_APP:
+              callback.onExpiredApp();
+              break;
+
+            default:
+              throw new IllegalArgumentException("Unsupported traffic state: " + mState);
+          }
+        }
+      });
+    }
+
+    private void iterateOverCallbacks(@NonNull Utils.Proc<TrafficCallback> proc)
+    {
+      for (TrafficCallback callback : mCallbacks)
+        proc.invoke(callback);
+    }
+  }
+
+  public interface TrafficCallback
+  {
+    void onEnabled();
+
```

Comment:
As there's no comments and signatures are quite obvious, I guess there is no need in blank lines between methods.